### DISCUSSION TITLE
SPI LCD & GPIO 12,13,14,15

### DISCUSSION THREAD

**Discussion by nradina** [pschatzmann/arduino-audiokit#60] [Q&A] [answered]
Hi all,

I'm trying to add an SPI LCD screen (with touch) to a project Im developing with this amazing AudioTools/AudioKit library.

I've reviewed the "[GPIO-Overview-by-Selected-Board](https://github.com/pschatzmann/arduino-audiokit/wiki/GPIO-Overview-by-Selected-Board)" link and other resources here.

I've also come across a [great video from Marcel ](https://youtu.be/r0af0DB1R68)about some hardware changes that may need to be made to enable SPI properly for a display.

I'm using the **number 5 ai_thinker (ES8388)** Audiokit board.

It seems there are the proper # and specific GPIO's available to take advantage of the SPI bus.
I _think_ the JTAG  header contains **GPIO_12,13,14,15.**

Yet I'm unsure how the JTAG header is configured with the dip-switches to allow access to those GPIO pins.

Here is my breakdown of my proposed pin assignments.

![screen_shot 2023-01-03 at 3 36 56 PM](https://user-images.githubusercontent.com/8020979/210437429-b6ad22de-5f59-407d-bafd-793405f3c085.jpg)

**Comment by nradina** [comment, marked as answer]
**SOLVED!** 

I spent some time today with a meter to track down what pins go where and how the dip-switches influence the path,
These are efforts to get a SPI TOUCH screen to work with the AudioKit and [Bodmer's excellent TFT_eSPI library](https://github.com/Bodmer/TFT_eSPI/)
My configuration with the dip-switches are: **0,0,0,1,1**

The JTAG header presents ESP32 pins as such (left to right):
**15,13,12,14.**

Big thanks to [this post ](https://forum.espuino.de/t/esp32-audio-kit-esp32-a1s/106/26)which helped me sort out the JTAG header.
Thanks to [this video by Marcel](https://youtu.be/r0af0DB1R68), I did the following:
I did unsolder the 0R resistors at R66,67,68,69,,70
And bridge pads R60,61,62,63,64.

In order to use the tact switches/buttons, I'll be adding 1.8K resistors at R55,56,57,58,59 .

I hope this helps others trying to get the Ai AudioKit board to work with a SPI display!

The pinout I chose and updated the TFT_eSPI "User_Select_Setup file" is:
![Screen Shot 2023-01-04 at 2 22 27 PM](https://user-images.githubusercontent.com/8020979/210633180-3c16c275-087a-45eb-8feb-2df099c5cd8e.png)

PIN breakouts:
![screen_shot 2023-01-04 at 2 12 46 PM](https://user-images.githubusercontent.com/8020979/210631520-538260ed-9768-4b72-80d6-f2a72d6055a0.jpg)

Photo of the dip-switches and pinouts:
![IMG_3414](https://user-images.githubusercontent.com/8020979/210632695-c568aa09-e154-4fe4-aacb-509aeefb3281.JPG)

**Comment by TangerineDreamer** [comment]
Thanks a lot to you nradina to spare your knowledge!!

i noticed your tft setup spares pins with the sdcard of the audiokit,so i wired CS->22 and DC->21
and now i have TFT  & SDcard working together and the SD player works too and send music to the headphones.
I have the ac101 version (6 in the settings) and does not give any sound to the HP,only to headphones.
I will make testings with touch buttons of the TFT, to override the hardware buttons of the audiokit cause
i don't need them anymore. i keeped the 01100 configuration of the multi-switch.
Thanks of course to pshatzmann for his library :)
